src/Libraries/PythonNodeModelsWpf/SharedCompletionProvider.cs

```diff
@@ -81,11 +95,21 @@ internal static IExternalCodeCompletionProviderCore FindMatchingCodeCompletionCo
         #endregion
 
         #region Methods
-        internal ICompletionData[] GetCompletionData(string code, bool expand = false)
+        internal ICompletionData[] GetCompletionData(string code, PythonEngineVersion engineVersion, bool expand = false)
         {
-            return this.providerImplementation?.GetCompletionData(code, expand).
-                Select(x => new IronPythonCompletionData(x)).ToArray();
+            if (engineVersion.Equals(PythonEngineVersion.IronPython2))
+            {
+                return this.providerImplementation?.GetCompletionData(code, expand).
+                    Select(x => new IronPythonCompletionData(x)).ToArray();
+            }
+            else if (engineVersion.Equals(PythonEngineVersion.CPython3))
+            {
+                return this.providerImplementation?.GetCompletionData(code, expand).
+                    Select(x => new IronPythonCompletionData(x)).ToArray();
```

Comment:
The providerImplementation will point to the appropriate completion provider core based on the Engine type. So that object would be different. Changed this to a single if statement.